### DISCUSSION TITLE
A question about undersampling

### DISCUSSION THREAD

**Discussion by Haozhoong** [facebookresearch/fastMRI#243] [General]
In the demo of transforms: '\fastmri\data\transforms.py', **UnetDataTransform** applies udnersampling mask in k space and then crops input to correct size in image domain and **VarNetDataTransform** does the same thing. However **MiniCoilTransform**  crops input to correct size in image domain and then applies udnersampling mask in k space.

There are zero-padding lines in phasing direction and they are recorded in h5py file at **attrs["padding_left"]**  and **attrs["padding_right"]**. Therefore, I think the two undersampling approaches above have different effective acceleration factors.  Which order is usd in the test dataset?

Thanks

**Comment by mmuckley** [comment]
Hello @Haozhoong, the 100% correct thing to do is what is done in `UnetDataTransform` and `VarNetDataTransform`. `MiniCoilTransform` was used in a sampling trajectory design paper to construct a miniature version of the full data for faster experimentation, but it doesn't do exactly what will occur in the scanner.

The test set applies the same undersampling as `UnetDataTransform` and `VarNetDataTransform`.

**Reply by Haozhoong** [reply]
OK, thanks!
